### PR TITLE
[TF] Fix incorrect use of weight decay

### PR DESCRIPTION
**Pull request by terrytangyuan**
Signed-off-by: terrytangyuan <terrytangyuan@gmail.com>

*Description of changes:* It was using exponential learning rate decay instead of weight decay.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by terrytangyuan** [comment]
Looks like there might be an issue with the CI since this error should not happen. Could you take a look? @mli

File: chapter_multilayer-perceptrons/kaggle-house-price.md [hunk 1]
```diff
@@ -674,7 +674,7 @@ will simplify uploading the results to Kaggle.
 #@tab all
 def train_and_pred(train_features, test_feature, train_labels, test_data,
                    num_epochs, lr, weight_decay, batch_size):
-    net = get_net()
+    net = get_net(weight_decay)
```
Comment (mli) [inline chapter_multilayer-perceptrons/kaggle-house-price.md:677]: `tab all` means it shares with mxnet and pytorch. if you only change the tf version, it will lead error for other frameworks.  I suggest don't change function apis

Comment (terrytangyuan) [reply chapter_multilayer-perceptrons/kaggle-house-price.md]: Got it. I didn't realize that this is in a `tab all` block. I have reverted the API change.

**Comment by mli** [comment]
Job d2l-en/PR-1025/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1025/
